ambry-tools/src/main/java/com.github.ambry/store/StoreCopier.java

```diff
@@ -237,12 +238,14 @@ public void close() throws IOException {
    * Copies data starting from {@code startToken} until all the data is copied.
    * @param startToken the {@link FindToken} to start copying from. Does not perform any duplication checks at
    *                   destination.
-   * @return the {@link FindToken} until which data has been copied.
+   * @return a {@link Pair} of the {@link FindToken} until which data has been copied and a {@link Boolean} indicating
+   * whether the source had problems that were skipped over - like duplicates ({@code true} indicates that there were).
    * @throws IOException if there is any I/O error while copying.
    * @throws StoreException if there is any exception dealing with the stores.
    */
-  public FindToken copy(FindToken startToken) throws IOException, StoreException {
-    FindToken lastToken = null;
+  public Pair<FindToken, Boolean> copy(FindToken startToken) throws IOException, StoreException {
```

Comment:
Is this method ever called with a `startToken` that is not a `newFindToken()`? If so, how is it expected to be used? Also, what does the comment for `startToken` that says "does not perform any duplication checks at destination" mean - since you do seem to make such a check? It seems to me that your earlier comment that says "until which data has been copied" made sense, because this method cannot accept *any* token (unlike replication which is idempotent) - because your `else` case at 277 will not be correct in that case.

Reply:
You raise a fair point - before all these changes I had envisioned being able to restart from a checkpoint but I realize now that it is not very useful and is probably more trouble that it is worth. I'll change this to not take in a `startToken` and update all the documentation.

Reply:
Actually, it is still useful for this to take a `startToken` but I will change the doucumentation.